autodiff/src/TermBuilder.cpp

```diff
@@ -173,54 +40,23 @@ shared_ptr<Term> TermBuilder::cos(shared_ptr<Term> arg) {
  *
  * @return A term describing the quadratic form
  */
-shared_ptr<Term> TermBuilder::quadform(shared_ptr<Term> x1, shared_ptr<Term> x2, shared_ptr<Term> a11,
-        shared_ptr<Term> a12, shared_ptr<Term> a21, shared_ptr<Term> a22) {
-    vector<shared_ptr<Term>> rest = {a22 * power(x2, 2)};
-    return make_shared<Sum>(a11 * power(x1, 2), (a12 + a21) * x1 * x2, rest);
-}
-
-shared_ptr<Term> TermBuilder::normalDistribution(shared_ptr<TVec> args, shared_ptr<TVec> mean, double variance) {
-    return exp((args - mean)->normSquared() * (-0.5 / variance)) * (1 / sqrt(2.0 * M_PI * variance));
+TermPtr TermBuilder::quadform(const TermPtr x1, const TermPtr x2, const TermPtr a11, const TermPtr a12, const TermPtr a21, const TermPtr a22)
+{
+    return a11 + x1->getOwner()->constPower(x1, 2) + (a12 + a21) * x1 * x2 + a22 * x2->getOwner()->constPower(x2, 2);
 }
 
-shared_ptr<Term> TermBuilder::gaussian(shared_ptr<TVec> args, shared_ptr<TVec> mean, double variance) {
-    return exp((args - mean)->normSquared() * (-0.5 / variance));
+TermPtr TermBuilder::sigmoid(const TermPtr arg, const TermPtr upperBound, const TermPtr lowerBound, const TermPtr mid, double steepness)
+{
+    return (upperBound - lowerBound) * (arg->getOwner()->sigmoid(arg, mid, steepness)) + lowerBound;
 }
 
-shared_ptr<Term> TermBuilder::sigmoid(shared_ptr<Term> arg, shared_ptr<Term> upperBound, shared_ptr<Term> lowerBound,
-        shared_ptr<Term> mid, double steepness) {
-    return (upperBound - lowerBound) * (make_shared<Sigmoid>(arg, mid, steepness)) + lowerBound;
+TermPtr TermBuilder::boundedValue(TermPtr arg, TermPtr leftBound, TermPtr rightBound)
+{
+    return arg->getOwner()->lessThan(leftBound, arg) & arg->getOwner()->lessThan(arg, rightBound);
```

Comment:
afaik "bounded" usually implies lessThanEqual.

Reply:
I'm fine with either, changing it.